internal/pkg/cli/local_run.go

```diff
@@ -292,23 +295,56 @@ func (o *localRunOpts) Execute() error {
 	}
 	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
 
-	err = o.runPauseContainer(context.Background(), containerPorts)
-	if err != nil {
-		return err
-	}
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Handle containers when a termination signal is received.
+	go func() {
+		sig := o.NotifyInterrupt()
+		log.Println("Received signal:", sig)
+		cancel()
+		errCh <- o.handleContainers()
+		// Wait for a max duration to allow containers to stop and get removed.
+		timeout := 30 * time.Second
+		select {
+		case <-time.After(timeout):
+			errCh <- fmt.Errorf("containers were not handled within the timeout %s", timeout)
+		default:
+			return
+		}
```

Comment:
i don't think this does anything 🤔 every single time `select` will pick default and always return - the error case will never happen

Reply:
Yes, this usually not happens, but what I though was for some reason if containers are not getting killed or not getting removed and If docker doesn't return any error that means we are not getting any error to the channel (I am not at all sure that even this happens). So just to make sure that case works by stopping the function after certain time. I have written it. We can remove it if it doesn't happen at all.